[18.0][MIG] stock_restrict_lot

StefanRijnhart opened the pull request:
Continued from https://github.com/OCA/stock-logistics-workflow/pull/1880

* Tests were refactored to reflect that Odoo two-step delivery now uses a push from outgoing to customer.
* Adjust implementation of rollup method to https://github.com/odoo/odoo/commit/f14d9b09b

Includes https://github.com/OCA/stock-logistics-workflow/pull/1560 and https://github.com/OCA/stock-logistics-workflow/pull/1773 that are not yet ported to 17.0

florian-dacosta committed:
Add module stock_restrict_lot

florian-dacosta committed:
Add module stock_restrict_lot

florian-dacosta committed:
Avoid copying restrict_lot_id field on stock move unless it is a split

florian-dacosta committed:
Avoid copying restrict_lot_id field on stock move unless it is a split

oca-travis committed:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

newtratip committed:
[FIX] stock_restrict_lot

newtratip committed:
[FIX] stock_restrict_lot

stock_restrict_lot 14.0.1.1.1

OCA-git-bot committed:
stock_restrict_lot 14.0.1.1.1

bizzappdev committed:
[IMP]stock_restrict_lot: migrate stock_restrict_lot module

[UPD] Update stock_restrict_lot.pot

[UPD] README.rst

petrus-v committed:
[FIX] stock_restrict_lot: do not merge stock move with different restrict_lot_id

Odoo tries to avoid extra stock.move while generating stock.picking from
pull/push rules. We must tell odoo to not merge stock.move if they are restricted
different lot.

To reproduce this behaviour using sale_order_lot_selection you have to:
* setup inventory with multi-step routes
* setup warehouse with two steps outgoing shipments
* receive products with different lot
* create a sale order with different quantities per lot
* validate the sale order

Before this PR stock.move in the internal stock.picking pick order are merged

After this PR we have one line per selected lot in internal stock.picking pick order

stock_restrict_lot 15.0.0.0.2

florian-dacosta committed:
[MIG] Migrate stock_restrict_lot to v16

florian-dacosta committed:
[IMP][stock_restrict_lot] Add lot in move list views

mt-software-de committed:
[IMP] compute product's outgoing and incoming qty by restrict_lot_id

badbole committed:
Added translation using Weblate (Croatian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/

mymage committed:
Added translation using Weblate (Italian)

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/es/

HekkiMelody committed:
[IMP] stock_restrict_lot: validate lot moved is correct

Kev-Roche committed:
[16.0][IMP] propagation of restrict_lot changes

metaminux committed:
[IMP] stock_restrict_lot : update restrict lot at stock move creation if in a chain and don't block an update if a move has already been done in the chain with this particular lot

jcadhoc committed:
[IMP] stock_restrict_lot: pre-commit auto fixes

jcadhoc committed:
[MIG] stock_restrict_lot: Migration to 17.0

grindtildeath committed:
[IMP] stock_restrict_lot: Add index to restrict_lot_id

Kev-Roche committed:
[IMP] restrict_lot propagation method

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: stock-logistics-workflow-17.0/stock-logistics-workflow-17.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-17-0/stock-logistics-workflow-17-0-stock_restrict_lot/it/

Kev-Roche committed:
[ADD] _rollup_not_cancelled_move_origs function

metaminux committed:
[FIX] stock_restrict_lot : actually assign and validate move instead of changing its state to done in tests

mav-adhoc committed:
[IMP] stock_restrict_lot: pre-commit auto fixes

emiliesoutiras committed:
[FIX] stock_restrict_lot : _rollup_not_cancelled_move_origs call itself

DavidJForgeFlow committed:
[REF] stock_restrict_lot: change AGPL-3 to LGPL-3

StefanRijnhart committed:
[FIX] stock_restrict_lot: fix outgoing quantity in test

The outgoing quantity is different depending on whether mrp is loaded, as the
demo data of that module adds an outgoing quantity of 3.
In the test, use a copy of the demo product to ensure there are no preexisting
stock operations.

Fixes
```
odoo.addons.stock_restrict_lot.tests.test_restrict_lot: FAIL: TestRestrictLot.test_compute_quantites
Traceback (most recent call last):
  File "/__w/stock-logistics-workflow/stock-logistics-workflow/stock_restrict_lot/tests/test_restrict_lot.py", line 254, in test_compute_quantites
    self.assertEqual(product.outgoing_qty, 2)
AssertionError: 5.0 != 2
```

StefanRijnhart commented:
/ocabot migration stock_restrict_lot

rjaraspearhead reviewed (approved):
functional review 👍

TDu commented on stock_restrict_lot/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1,4 @@
+This module add a field to restrict a stock move to a specific lot. It
```
```suggestion
This module adds a field to restrict a stock move to a specific lot. It
```

StefanRijnhart replied:
Thanks, done.

StefanRijnhart commented:
@florian-dacosta as the maintainer of this module, do you want to have a look as well and then maybe merge?

florian-dacosta commented:
Thanks for the ping and the migration @StefanRijnhart 
The push process had been implemented in v16 actually, between other stuff, here : https://github.com/OCA/stock-logistics-workflow/pull/1560 
It was not forward ported though.
If you think it simple enough to include it in this PR and you have the time to do it, it is great (https://github.com/OCA/stock-logistics-workflow/pull/1773 also comes as a fix)

If not, I am ok to merge it as it is anyway, @Kev-Roche should be able to do the forward port in a separated PR.
(Although I have not the power to merge it, I think the maintainer stuff does work only if the module is already in the targeted branch)

StefanRijnhart commented:
@florian-dacosta thanks for the pointers, I'll check those improvements.

StefanRijnhart commented:
@florian-dacosta I integrated the PRs you mentioned.

mav-adhoc committed:
[MIG] stock_restrict_lot: Migration to 18.0

Tests were refactored to reflect that Odoo two-step delivery now uses a push
from outgoing to customer.

Adjust implementation of rollup method to https://github.com/odoo/odoo/commit/f14d9b09b

Kev-Roche reviewed (approved):
LGTM, thank you !

florian-dacosta commented:
I'll try to merge as maintainer, but I don't think I can !

florian-dacosta commented:
/ocabot merge nobump

StefanRijnhart commented:
@florian-dacosta Thanks, and good to know that module maintainers can merge migration PRs!
